adapters/infoawarebidder.go

```diff
@@ -0,0 +1,160 @@
+package adapters
+
+import (
+	"fmt"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+// InfoAwareBidder wraps a Bidder to ensure all requests abide by the capabilities and
+// media types defined in the static/bidder-info/{bidder}.yaml file.
+//
+// It adjusts incoming requests in the following ways:
+//   1. If App or Site traffic is not supported by the info file, then requests from
+//      those sources will be rejected before the delegate is called.
+//   2. If a given MediaType is not supported for the platform, then it will be set
+//      to nil before the request is forwarded to the delegate.
+//   3. Any Imps which have no MediaTypes left will be removed.
+//   4. If there are no valid Imps left, the delegate won't be called at all.
+type InfoAwareBidder struct {
+	Bidder
+	info parsedBidderInfo
+}
+
+// BuildInfoAwareBidder wraps a bidder to enforce site, app, and media type support.
+func BuildInfoAwareBidder(bidder Bidder, info config.BidderInfo) Bidder {
+	return &InfoAwareBidder{
+		Bidder: bidder,
+		info:   parseBidderInfo(info),
+	}
+}
```

Comment:
Changed the builder here slightly for a more uniform naming convention. Rest of file is unchanged, just renamed.